locale/en/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,457 @@
+
+Note that the number of sub-Tasks completed is not a useful metric for the throughput of the Worker Pool.
+Instead, concern yourself with the number of *Tasks* completed.
+
+### Avoiding Task partitioning
+Recall that the purpose of Task partitioning is to minimize the variation in Task times.
+If you can distinguish between shorter Tasks and longer Tasks (e.g. summing an array vs. sorting an array), you could create one Worker Pool for each class of Task.
+Routing shorter Tasks and longer Tasks to separate Worker Pools is another way to minimize Task time variation.
+
+In favor of this approach, partitioning Tasks incurs overhead (the costs of creating a Worker Pool Task representation and of manipulating the Worker Pool queue), and avoiding partitioning saves you the costs of additional trips to the Worker Pool.
+It also keeps you from making mistakes in partitioning your Tasks.
+
+The downside of this approach is that Workers in all of these Worker Pools will incur space and time overheads and will compete with each other for CPU time.
+Remember that each CPU-bound Task makes progress only while it is scheduled.
+As a result, you should only consider this approach after careful analysis.
+
+### Conclusions
+Whether you use only the Node Worker Pool or maintain separate Worker Pool(s), you should optimize the Task throughput of your Pool(s).
+
+To do this, minimize the variation in Task times by using Task partitioning.
+
+##  The risks of npm modules
+Node developers benefit tremendously from the [npm ecosystem](https://www.npmjs.com/), with hundreds of thousands of modules offering functionality to accelerate your development process.
+
+However, the vast majority of these modules are written by third-party developers and undergo no formal verification process.
+A developer using an npm module should be concerned about two things, though the latter is frequently forgotten.
+1. Does it honor its APIs?
+2. Might its APIs block the Event Loop or a Worker?
+Many modules make no effort to indicate the cost of their APIs, to the detriment of the community.
+
+For simple APIs you can estimate the cost of the APIs; the cost of string manipulation isn't hard to fathom.
+But in many cases it's unclear how much an API might cost.
+
+*If you are calling an API that might do something expensive, ask the developers to document its cost, or examine the source code yourself (and submit a PR documenting the cost).*
+
+Remember, even if the API is asynchronous, you don't know how much time it might spend on a Worker or on the Event Loop in each of its partitions.
+For example, suppose in the `asyncAvg` example given above, each call to the helper function summed *half* of the numbers rather than one of them.
+Then this function would still be asynchronous, but the cost of each partition would be `O(n)`, not `O(1)`, making it much less safe to use for arbitrary values of `n`.
+
+## Conclusion
+Node has two types of threads: one Event Loop and `k` Workers.
```

Comment:
What about the JavaScript thread?

Reply:
I am attempting to refer to the "JavaScript thread" as the Event Loop throughout, to contrast it with the Worker Pool (threadpool) threads.

See the section "What code runs on the Event Loop?"

"In summary, the Event Loop executes the JavaScript callbacks registered for events, and is also responsible for fulfilling non-blocking asynchronous requests like network I/O."

Let me know if this is not clear or if you have alternative suggestions for phrasing.